pkg/reconciler/serverlessservice/global_resync_test.go

```diff
@@ -0,0 +1,213 @@
+/*
+   Copyright 2019 The Knative Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package serverlessservice
+
+import (
+	"testing"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	ctrl "github.com/knative/pkg/controller"
+	logtesting "github.com/knative/pkg/logging/testing"
+	fakeclientset "github.com/knative/serving/pkg/client/clientset/versioned/fake"
+	informers "github.com/knative/serving/pkg/client/informers/externalversions"
+	preconciler "github.com/knative/serving/pkg/reconciler"
+
+	autoscalingapi "k8s.io/api/autoscaling/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	fakedynamic "k8s.io/client-go/dynamic/fake"
+	kubeinformers "k8s.io/client-go/informers"
+	fakekubeclientset "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/scale"
+	"k8s.io/client-go/tools/record"
+
+	. "github.com/knative/pkg/reconciler/testing"
+	"github.com/knative/serving/pkg/reconciler/serverlessservice/resources/names"
+	. "github.com/knative/serving/pkg/reconciler/testing"
+)
+
+type mockScale struct {
+	scale.ScaleInterface
```

Comment:
we don't use this anymore in https://github.com/knative/serving/pull/3893

Reply:
yeah didn't merge yet though :P

Reply:
Look again 😈

Reply:
🛤

Reply:
👨‍🎤